src/xrpld/app/ledger/detail/LedgerMaster.cpp

```diff
@@ -1445,11 +1445,21 @@ LedgerMaster::findNewLedgersToPublish(
             }
 
             // Did we acquire the next ledger we need to publish?
-            if (ledger && (ledger->info().seq == pubSeq))
+            if (ledger && (ledger->info().seq >= pubSeq))
             {
                 ledger->setValidated();
                 ret.push_back(ledger);
-                ++pubSeq;
+                if (ledger->info().seq > pubSeq)
+                {
+                    JLOG(m_journal.info())
+                        << "ready to publish " << ledger->info().seq
+                        << ", but skipping " << pubSeq
+                        << " because it has not been acquired.";
+                }
```

Comment:
Is it possible to write a unit test to exercise this new execution branch ?

Reply:
There's no coverage of this function or its caller. It's caller's caller is tested twice, but for entirely different reasons. This function's caller, LedgerMaster::doAdvance(), does a lot of things, none of which are currently tested, let alone this particular tweak.